app/controllers/sites_controller.rb

```diff
@@ -18,7 +18,14 @@ def show
   end
 
   def update
-    if @site.update(site_params)
+    @site.attributes = site_params
+    if(@site.terms_of_use_content_changed?)
+      id = current_user.id
+      users = User.where("id != ?", id)
+      users.update_all(terms_of_use_accepted: false)
```

Comment:
This is one possible way of course. What would you think about a solution that uses `terms_accepted_at (DateTime)` on a user model - this way you would simply compare `updated_at` on Site with the `terms_accepted_at` on a model to check whether the user has accepted the updated terms.

Reply:
Yeah this solution is nice to. I following instructions in task (https://github.com/panterch/future_kids/issues/317)
And it depend ... if we need to save date of accepted it will be better your solution. but if we wouldn't need it, i think solution with boolean is better .... for example when we need check in DB who accept and who not it will be more readeble when we have this data. Of course we can have both or it is just simple sql query to show it but if we no need date of accept i think it is useles.

Reply:
Since you introduced terms_of_use_accepted_date the boolean field terms_of_use_accepted is a duplicate. I'd would stay with terms_of_use_accepted_date on the user and simply add a virtual field on user:

def terms_of_use_accepted
  user.terms_of_use_accepted_date < site.terms_of_use_accepted_date 
end


Then you don't have to update all database rows when the terms change.

Reply:
yes, good tip. I changed it..